php/class-support.php

```diff
@@ -0,0 +1,226 @@
+	 * @param string $action   The action to handle.
+	 * @param array  $post_ids Post ID's to action.
+	 *
+	 * @return string
+	 */
+	public function add_to_report( $location, $action, $post_ids ) {
+		if ( 'cloudinary-report' === $action ) {
+			$items = get_option( self::REPORT_KEY, array() );
+			foreach ( $post_ids as $id ) {
+				if ( ! in_array( $id, $items, true ) ) {
+					$items[] = $id;
+				}
+			}
+			update_option( self::REPORT_KEY, $items, false );
+		}
+
+		return $location;
+	}
+
+	/**
+	 * Add an inline action for adding to report.
+	 *
+	 * @param array    $actions All actions.
+	 * @param \WP_Post $post    The current post object.
+	 *
+	 * @return array
+	 */
+	public function add_inline_action( $actions, $post ) {
+
+		if ( 'on' === $this->settings->get_value( 'enable_support' ) ) {
+
+			$screen = get_current_screen();
+			if ( $screen && 'upload' === $screen->id ) {
+
+				$args = array(
+					'action'   => 'cloudinary-report',
+					'media[]'  => $post->ID,
+					'_wpnonce' => wp_create_nonce( 'bulk-media' ),
+				);
+
+			} else {
+				$args = array(
+					'action'   => 'cloudinary-report',
+					'post[]'   => $post->ID,
+					'_wpnonce' => wp_create_nonce( 'bulk-posts' ),
+				);
+			}
+			$action_url                    = add_query_arg( $args, '' );
+			$title                         = esc_html__( 'Add to Cloudinary Report', 'cloudinary' );
+			$actions['cloudinary-support'] = sprintf(
+				'<a href="%1$s" aria-label="%2$s">%2$s</a>',
+				$action_url,
+				$title
+			);
+		}
+
+		return $actions;
+	}
+
+	/**
+	 * Setup the component.
+	 */
+	public function setup() {
+		if ( 'on' === $this->settings->get_value( 'enable_support' ) ) {
+			add_action( 'add_meta_boxes', array( $this, 'image_meta_viewer' ) );
+		}
+	}
+
+	/**
+	 * Init the report by clearing and preparing the report options.
+	 *
+	 * @param mixed   $new_value     The new value.
+	 * @param mixed   $current_value The current value.
+	 * @param Setting $setting       The setting object.
+	 *
+	 * @return mixed
+	 */
+	public function init_reporting( $new_value, $current_value, $setting ) {
+		delete_option( self::REPORT_KEY );
+
+		return $new_value;
+	}
+
+	/**
+	 * Add Meta view meta box.
+	 */
+	public function image_meta_viewer() {
+		$screen = get_current_screen();
+		if ( ! $screen instanceof \WP_Screen || 'attachment' !== $screen->id ) {
+			return;
+		}
+
+		add_meta_box(
+			'meta-viewer',
+			__( 'Cloudinary Metadata viewer', 'cloudinary' ),
+			array( $this, 'render' )
+		);
+	}
+
+	/**
+	 * Render the metabox content.
+	 *
+	 * @param \WP_Post $post The post.
+	 */
+	public function render( $post ) {
+		if ( 'attachment' === $post->post_type ) {
+			$meta = wp_get_attachment_metadata( $post->ID );
+
+			$args = array(
+				'type'       => 'tag',
+				'element'    => 'pre',
+				'attributes' => array(
+					'style' => 'overflow:auto;',
+				),
+				'content'    => wp_json_encode( $meta, JSON_PRETTY_PRINT ),
```

Comment:
@DavidCramer 

What do you think if we also add the flag for `JSON_UNESCAPED_SLASHES`, as in

```suggestion
				'content'    => wp_json_encode( $meta, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES ),
```

This removes the backslash, and I haven't found any drawbacks.
I tested this in all instances where we output this to the screen, but also on the downloaded file.

Reply:
@pereirinha - Good suggestion!